mlflow/server/js/src/experiment-tracking/components/ExperimentListView.js

```diff
@@ -19,22 +30,63 @@ import { IconButton } from '../../common/components/IconButton';
 export class ExperimentListView extends Component {
   static propTypes = {
     activeExperimentIds: PropTypes.arrayOf(PropTypes.string).isRequired,
-    experiments: PropTypes.arrayOf(Experiment).isRequired,
     history: PropTypes.object.isRequired,
+    // All below depend on connect
+    dispatchSearchExperimentsApi: PropTypes.func.isRequired,
+    dispatchSearchInput: PropTypes.func.isRequired,
+    experiments: PropTypes.arrayOf(PropTypes.object),
+    searchInput: PropTypes.string.isRequired,
+    previousSearchInput: PropTypes.string.isRequired,
+    nextPageToken: PropTypes.string,
+    innerHeight: PropTypes.number.isRequired,
   };
 
   state = {
     hidden: false,
-    searchInput: '',
     showCreateExperimentModal: false,
     showDeleteExperimentModal: false,
     showRenameExperimentModal: false,
     selectedExperimentId: '0',
     selectedExperimentName: '',
+    checkedKeys: [],
   };
 
   handleSearchInputChange = (event) => {
-    this.setState({ searchInput: event.target.value });
+    this.props.dispatchSearchInput(event.target.value);
+  };
+
+  handleLoadMore = (event) => {
+    let params = {};
+    const currentInput = this.props.searchInput;
+    const previousInput = this.props.previousSearchInput;
+    // Leave the filter off if we have a blank string.
+    if (currentInput !== '') {
+      params = { ...params, filter: `name LIKE '%${currentInput}%'` };
+    }
+    // Use a next page if the input was not altered to get more.
+    if (currentInput === previousInput) {
+      params = { ...params, pageToken: this.props.nextPageToken };
+    }
+    // Need to dispatch both to make the prior and current match on
+    // multiple searches of the same input.
+    this.props.dispatchSearchInput(currentInput);
+    this.props.dispatchSearchExperimentsApi(params);
+  };
+
+  // TODO there is probably a better way to make this dynamic
+  // but it must be a number, so no vh.
+  containerHeight = Math.abs(this.props.innerHeight * 0.9);
```

Comment:
it would be great to avoid magic values here - maybe can we employ a super simple [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) here to get the exact value for container height?

Reply:
Implemented a basic observer. It still uses a percentage of the root but seems much better than before. Is this along the lines of what you were thinking?